code/ai/aicode.cpp

```diff
@@ -13044,9 +13048,20 @@ void ai_manage_shield(object *objp, ai_info *aip)
 		//	Scale between 1x and 3x based on ai_class (SUSHI: only if autoscale is on)
 		if (aip->ai_class_autoscale)
 		{
-			if (The_mission.ai_profile->flags[AI::Profile_Flags::Adjusted_AI_class_autoscale])
-				delay = delay + delay * (float)(3 * (ai_get_autoscale_index(Num_ai_classes) - ai_get_autoscale_index(aip->ai_class) - 1) / (ai_get_autoscale_index(Num_ai_classes) - 1));
-			else
+			int autoscale_index = ai_get_autoscale_index(Num_ai_classes);
```

Comment:
The `autoscale_index` variable should be renamed `autoscale_num_classes` for clarity.  Note that both the class index and the number of classes are passed to `ai_get_autoscale_index`.